src/KeyedFrames.jl

```diff
@@ -104,7 +99,7 @@ Base.setindex!(kf::KeyedFrame, value, ind...) = setindex!(frame(kf), value, ind.
 function _kf_getindex(kf::KeyedFrame, index...)
     # If indexing by column, some keys might be removed.
     df = getindex(frame(kf), index...)
-    return KeyedFrame(df, intersect(names(df), keys(kf)))
+    return KeyedFrame(DataFrame(df), intersect(names(df), keys(kf)))
```

Comment:
Why this change?
I guess to make sure `SubDataFrames` are copied into new `DataFrames`

Also note that in the next release of DataFrames.jl this will become a copying operation.
Even if `df` is a actual `DataFrame` (rather than a `SubDataFrame`)
Which is probably the desired behavour and the reason for doing this anyway.
Since we want a copy iin tghe case when someone might have multiple aliases for the same `DataFrame`

Reply:
`@test isequal(kf1[1, [:a, :b]], KeyedFrame(DataFrame(; a=1, b=2), [:a, :b]))` fails with the following exception 
```
getindex: Error During Test at /Users/maryjoramos/.julia/dev/KeyedFrames/test/runtests.jl:109
  Test threw exception
  Expression: isequal(kf1[1, [:a, :b]], KeyedFrame(DataFrame(; a=1, b=2), [:a, :b]))
  MethodError: no method matching KeyedFrame(::DataFrameRow{DataFrame,DataFrames.SubIndex{DataFrames.Index,Array{Int64,1},Array{Int64,1}}}, ::Array{Symbol,1})
  Closest candidates are:
    KeyedFrame(!Matched::DataFrame, ::Array{#s12,1} where #s12<:Symbol) at /Users/maryjoramos/.julia/dev/KeyedFrames/src/KeyedFrames.jl:13
```

Reply:
Right.
I think this is the best way.
I think making something into a `KeyedFrames` should copy.
Since we mostly call `join` on `KeyedFrames`,
and I don't think doing that to a SubDataFrame should work.

I think this is good as is.